### ISSUE TITLE
Image does not appear to have ESI tags enabled

### ISSUE THREAD

**Issue by shuffless** [million12/docker-varnish#9]
We are running this Docker Image on a Plesk environment with a custom Magento 2 VCL mapped for starting the image. Everything works well except for ESI includes. A particular case is the menu of a website is not available with Varnish on and a tag like the one below appears instead of the menu

`<esi:include src="http://www.domain.com/page_cache/block/esi/blocks/..."></esi:include>
`

The problem seems to be in the DAEMON_OPTS section of the Image as we discussed with some developers and [they suggested we add this](http://docs.swissuplabs.com/m2/extensions/navigationpro/troubleshooting/missing-top-navigation-when-varnish-is-used/) in DAEMON_OPTS section. 

```
    -p feature=+esi_ignore_other_elements \
    -p feature=+esi_disable_xml_check \
    -p feature=+esi_ignore_https \
    -p http_resp_hdr_len=16k \
    -p http_resp_size=64k \
    -p workspace_backend=64k \
```

We tried to add DAEMON_OPTS as a parameter for Docker but it is not working 

![69bdf074-6be3-494e-94fd-f356459e5d0f](https://user-images.githubusercontent.com/4831785/129191168-5c9be7d9-079a-46e9-ad7c-3f61c9a70c26.jpg)

We also searched for the option to enable them directly in the Docker Image config files and we found this option in 2 locations 

Location 1 
`./etc/varnish/varnish.params
`

For this we added this line

`DAEMON_OPTS="-p feature=+esi_ignore_other_elements -p feature=+esi_disable_xml_check -p feature=+esi_ignore_https -p http_resp_hdr_len=16k -p http_resp_size=64k -p workspace_backend=64k"`

Location 2

`./usr/lib/systemd/system/varnish.service
`

For this the config we tried to include the options directly in the ExecStart lines like so

Original config

```
ExecStart=/usr/sbin/varnishd \
        -P /var/run/varnish.pid \
        -f $VARNISH_VCL_CONF \
        -a ${VARNISH_LISTEN_ADDRESS}:${VARNISH_LISTEN_PORT} \
        -T ${VARNISH_ADMIN_LISTEN_ADDRESS}:${VARNISH_ADMIN_LISTEN_PORT} \
        -S $VARNISH_SECRET_FILE \
        -u $VARNISH_USER -g $VARNISH_GROUP \
        -s $VARNISH_STORAGE \
        $DAEMON_OPTS
```

New Config

```
ExecStart=/usr/sbin/varnishd \
        -P /var/run/varnish.pid \
        -f $VARNISH_VCL_CONF \
        -a ${VARNISH_LISTEN_ADDRESS}:${VARNISH_LISTEN_PORT} \
        -T ${VARNISH_ADMIN_LISTEN_ADDRESS}:${VARNISH_ADMIN_LISTEN_PORT} \
        -S $VARNISH_SECRET_FILE \
        -u $VARNISH_USER -g $VARNISH_GROUP \
        -s $VARNISH_STORAGE \
    -p feature=+esi_ignore_other_elements \
    -p feature=+esi_disable_xml_check \
    -p feature=+esi_ignore_https \
    -p http_resp_hdr_len=16k \
    -p http_resp_size=64k \
    -p workspace_backend=64k \
        $DAEMON_OPTS
```

Basically $DAEMON_OPTS should have included the options from the previous file yet none of the options are working. 

The VCL contains the proper esi tag like so

```
sub vcl_backend_response {

    set beresp.grace = 3d;

    if (beresp.http.content-type ~ "text") {
       **set beresp.do_esi = true;**
    }

```

**Comment by shuffless** [comment]
The problem resided elsewhere.
